src/main/scala/io/iohk/ethereum/faucet/FaucetSupervisor.scala

```diff
@@ -0,0 +1,47 @@
+package io.iohk.ethereum.faucet
+
+import akka.actor.{ActorRef, ActorSystem, OneForOneStrategy, SupervisorStrategy}
+import akka.pattern.{BackoffOpts, BackoffSupervisor}
+import io.iohk.ethereum.faucet.FaucetHandler.WalletException
+import io.iohk.ethereum.faucet.jsonrpc.WalletService
+import io.iohk.ethereum.utils.Logger
+
+import scala.concurrent.duration._
+
+object FaucetSupervisor {
+  val name = "FaucetSupervisor"
+}
+
+class FaucetSupervisor(walletRpcClient: WalletService, config: FaucetConfig, shutdown: () => Unit)(implicit
```

Comment:
I think is better to just pass the Handler, i mean, why the supervisor needs to know about the WalletService?